### DISCUSSION TITLE
Include column number in the AST nodes

### DISCUSSION THREAD

**Discussion by andreabergia** [mozilla/rhino#1719] [Ideas]
Rhino's AST nodes currently include the row number of the original source file. However, they do not include the column number.

The row number is used in various places:
- for stack traces, for interpreter and compiled mode
- for the debugger interface

We propose to add column numbers because they are required to implement correctly source maps. We have an internal implementation at ServiceNow, to process transpiled files, that works in this way:

- rhino will execute the transpiled file, which comes with an attached source map
- we have added column numbers to ast nodes
- when we generate the line number table (for compiled mode) or `LINE` bytecodes (for interpreter mode) we go through the source map to associate the current node with the location in the original file. So the line number table will refer to the _original_ source file, not the transpiled one that rhino is actually executing.

We have plans to redo part of our code related to source maps and to upstream these changes, but as a first step, we would like to add column numbers to all the AST nodes, to simplify our maintenance.

Are there any objections? If not, we can try to create a PR for this in the next few weeks.

**Comment by p-bakker** [comment]
Related to #1308

**Comment by gbrail** [comment]
If you all can figure out how to get column numbers in there, I'm sure lots of people would benefit!

**Reply by andreabergia** [reply]
Working on it in https://github.com/mozilla/rhino/pull/1724
